syntax = "proto3";
option go_package = "rpc_function/calculator_rpc";
package calculator;

// Pesan berupa input bilangan
message CalcRequest {
  float operand1 = 1;
  float operand2 = 2;
}

// Pesan berupa response output server untuk penjumlahan
message CalcAddResponse {
  float result = 1;
}

// Pesan berupa response output server untuk pengurangan
message CalcSubtractResponse {
  float result = 1;
}

// Pesan berupa response output server untuk pembagian
message CalcDivideResponse {
  float result = 1;
  bool zero_div = 2;
}

// Pesan berupa response output server untuk perkalian
message CalcMultiplyResponse {
  float result = 1;
}

// Definisi layanan gRPC
service CalcService {
  rpc CalcAdd (CalcRequest) returns (CalcAddResponse);
  rpc CalcSubtract (CalcRequest) returns (CalcSubtractResponse);
  rpc CalcDivide (CalcRequest) returns (CalcDivideResponse);
  rpc CalcMultiply (CalcRequest) returns (CalcMultiplyResponse);
}
